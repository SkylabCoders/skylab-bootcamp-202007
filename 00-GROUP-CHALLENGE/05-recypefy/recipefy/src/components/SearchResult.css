.result-section {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  margin: 10%;
}

.result-section h2 {
  font-size: 30px;
}

.result-section .recipe__text,
.result-section .recipe-card {
  width: 100%;
}

.result-section .recipe__text h2 {
  margin-bottom: 10%;
}

.result-section .body__box {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
}

.result-section .body__box a {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

@media only screen and (min-width: 600px) {
  .result-section .body__box {
    max-width: 800px;
  }
  .result-section .body__box a {
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
        -ms-flex-direction: row;
            flex-direction: row;
  }
}
/*# sourceMappingURL=SearchResult.css.map */