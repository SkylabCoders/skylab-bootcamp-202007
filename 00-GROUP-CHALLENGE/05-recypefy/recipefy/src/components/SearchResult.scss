.result-section {
	display: flex;
	flex-direction: column;
	align-items: center;
	margin: 10%;

	h2 {
		font-size: 30px;
	}

	.recipe__text,
	.recipe-card {
		width: 100%;
	}

	.recipe__text h2 {
		margin-bottom: 10%;
	}

	.body__box {
		display: flex;
		flex-direction: column;

		a {
			display: flex;
			flex-direction: column;
			align-items: center;
		}
	}
}

@media only screen and (min-width: 600px) {
	.result-section {

		.body__box {
			max-width: 800px;

			a {
				flex-direction: row;
			}
		}

	}
}