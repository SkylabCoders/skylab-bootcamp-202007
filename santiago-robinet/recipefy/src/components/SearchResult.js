import React from 'react';


function SearchResult(){
    return(
        <>
          <p> I'm da SearchResult!! And I work!!</p>
        </>
    );
}

export default SearchResult;