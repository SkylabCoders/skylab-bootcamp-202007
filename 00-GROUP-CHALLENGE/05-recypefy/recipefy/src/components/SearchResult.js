import React, { useEffect, useState } from 'react';
import recipeStore from '../stores/RecipeStore';
import { loadRecipe } from '../actions/RecipeAction';
import RecipeCard from './RecipeCard.component';
import './SearchResult.css';

function SearchResult() {
	const [searchListResult, setSearchListResult] = useState(
		recipeStore.getRecipes
	);

	useEffect(() => {
		recipeStore.addChangeListener(onChange);
		if (searchListResult.length === 0) {
			loadRecipe();
		}
		return () => recipeStore.removeChangeListener;
	}, [searchListResult.length]);

	function onChange() {
		setSearchListResult(recipeStore.getRecipes);
	}
	return (
		<div className="result-section">
			<h2>YOUR RESULTS!</h2>
			{searchListResult.map((recipe) => (
				<RecipeCard title={recipe.title} />
			))}
		</div>
	);
}

export default SearchResult;
