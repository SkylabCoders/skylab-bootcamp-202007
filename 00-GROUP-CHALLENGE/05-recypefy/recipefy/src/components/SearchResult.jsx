import React, { useEffect, useState } from 'react';
import recipeStore from '../stores/RecipeStore';
import RecipeCard from './RecipeCard.component';
import './SearchResult.scss';

function SearchResult() {
	const [searchListResult, setSearchListResult] = useState(
		recipeStore.getRecipes
	);
	const [chargeTime, setChargeTime] = useState(false);

	useEffect(() => {
		recipeStore.addChangeListener(onChange);
		chargeList();
		if (searchListResult.length != 0) {
			resetTime();
		}
		return () => recipeStore.removeChangeListener;
	}, [searchListResult.length, chargeTime]);

	function onChange() {
		setSearchListResult(recipeStore.getRecipes);
	}

	function chargeList() {
		setTimeout(() => {
			setChargeTime(true);
		}, 3000);
	}
	function resetTime() {
		setChargeTime(false);
	}
	return (
		<div className="result-section body-content">
			{searchListResult && (
				<>
					<h2>YOUR RESULTS!</h2>
					{searchListResult.map((recipe) => (
						<RecipeCard title={recipe.title} key={recipe.title} />
					))}
				</>
			)}
			{searchListResult.length === 0 && (
				<>
					{!chargeTime && (
						<>
							<img
								className="loading"
								alt="loading"
								src="https://media1.tenor.com/images/8a1df0caea44830aa0ecf99ac223c747/tenor.gif?itemid=15085886"
							></img>
							<p>loading...</p>
						</>
					)}
					{chargeTime && (
						<>
							<h2 className="box__title">Your search have no results!</h2>
							<h3>Try again</h3>
							<p>
								Perhaphs is an error with the search or your search don't have
								any matches.
							</p>
						</>
					)}
				</>
			)}
		</div>
	);
}

export default SearchResult;
