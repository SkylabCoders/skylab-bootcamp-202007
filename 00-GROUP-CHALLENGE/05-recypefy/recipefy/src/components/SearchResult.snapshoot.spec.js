import renderer from 'react-test-renderer';
import SearchResult from './SearchResult';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
describe('SearchResult snapshot', () => {
    const tree = renderer.create(
        <BrowserRouter>
            <SearchResult />
        </BrowserRouter>
    );
    it('should match', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    })
})