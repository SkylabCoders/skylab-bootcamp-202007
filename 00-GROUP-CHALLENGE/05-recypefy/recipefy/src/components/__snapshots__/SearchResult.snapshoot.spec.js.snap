// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult snapshot should match 1`] = `
<div
  className="result-section body-content"
>
  <h2>
    YOUR RESULTS!
  </h2>
  <img
    alt="loading"
    className="loading"
    src="https://media1.tenor.com/images/8a1df0caea44830aa0ecf99ac223c747/tenor.gif?itemid=15085886"
  />
  <p>
    loading...
  </p>
</div>
`;
